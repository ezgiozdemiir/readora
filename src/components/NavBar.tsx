import React from 'react';
import { Container, Group, Button } from '@mantine/core';
import { Link } from 'react-router-dom';

const NavBar: React.FC = () => {
  return (
    <Container>
      <Group >
        <Button component={Link} to="/" variant="light" color="blue">Home</Button>
      </Group>
    </Container>
  );
};

export default NavBar;
