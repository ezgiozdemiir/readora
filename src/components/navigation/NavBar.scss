.navbar{
padding: 0.5rem;
position:fixed;
background-color: #ffff;
width: 100vw !important;
z-index: 10;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); 
   img{
    width: 8rem;
} 
}

.m_7485cace{
    --container-size-md: 100vw !important
}