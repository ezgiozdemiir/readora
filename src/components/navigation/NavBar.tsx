import React from 'react'
import { Container, Group, Button } from '@mantine/core'
import { Link } from 'react-router-dom'
import logo from '../../assets/Readora.png'
import './NavBar.scss'

const NavBar: React.FC = () => {
    return (
        <Container className="navbar">
            <Group>
                <img src={logo} alt="readora" />
                <Button component={Link} to="/" variant="light" color="blue">
                    Home
                </Button>
                <Button
                    component={Link}
                    to="/profile"
                    variant="light"
                    color="blue"
                >
                    Profile
                </Button>
                <Button
                    component={Link}
                    to="/login"
                    variant="light"
                    color="blue"
                >
                    Login
                </Button>
            </Group>
        </Container>
    )
}

export default NavBar
