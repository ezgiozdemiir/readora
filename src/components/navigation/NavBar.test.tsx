import { render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { MemoryRouter } from 'react-router-dom';
import NavBar from './NavBar';

//Logo'yu mocklamayınca hata veriyor, Jest image dosyalarını import edemiyor, dosyayı çözümleyemiyor. Burada aslında import etme 'mocked-logo.png' stringini döndür diyoruz.
jest.mock('../../assets/Readora.png', () => 'mocked-logo.png');

test('NavBar matches snapshot', () => {
  const { asFragment } = render(
    <MantineProvider>
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    </MantineProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});