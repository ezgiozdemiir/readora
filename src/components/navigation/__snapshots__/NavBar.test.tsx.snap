// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NavBar matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="navbar m_7485cace mantine-Container-root"
  >
    <div
      class="m_4081bf90 mantine-Group-root"
      style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <img
        alt="readora"
        src="mocked-logo.png"
      />
      <a
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-discover="true"
        data-variant="light"
        href="/"
        style="--button-bg: var(--mantine-color-blue-light); --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Home
          </span>
        </span>
      </a>
      <a
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-discover="true"
        data-variant="light"
        href="/profile"
        style="--button-bg: var(--mantine-color-blue-light); --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Profile
          </span>
        </span>
      </a>
      <a
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-discover="true"
        data-variant="light"
        href="/login"
        style="--button-bg: var(--mantine-color-blue-light); --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Login
          </span>
        </span>
      </a>
      <button
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="light"
        style="--button-bg: var(--mantine-color-red-light); --button-hover: var(--mantine-color-red-light-hover); --button-color: var(--mantine-color-red-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Logout
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
